### PR TITLE
Merge tectonic with rewards

### PR DESCRIPTION
**Pull request by BenReilly19**

### CODE REVIEW

**Comment by BenReilly19** [comment]
*** Some liquidation amounts are 0 on pools but this will be corrected as indexing progresses

File: subgraphs/compound-forks/protocols/tectonic/src/mapping.ts [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+  let borrowCompPerDay = BIGINT_ZERO;
+
+  // Tonic speeds are the same for supply/borrow side
+  let tryCompSpeed = troller.try_tonicSpeeds(event.address);
+  supplyCompPerDay = tryCompSpeed.reverted
+    ? BIGINT_ZERO
+    : tryCompSpeed.value.times(blocksPerDay);
+  borrowCompPerDay = supplyCompPerDay;
+
+  // get TONIC price
```
Comment (dmelotik) [inline subgraphs/compound-forks/protocols/tectonic/src/mapping.ts:324]: remove this commented out code if not using

File: subgraphs/compound-forks/protocols/tectonic/src/mapping.ts [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+      .div(exponentToBigDecimal(rewardDecimals));
+  } else if (event.block.number.toI32() > 1337194) {
+    let compMarket = Market.load(tTONICAddress);
+    if (!compMarket) {
+      log.warning("[updateRewards] Market not found: {}", [tTONICAddress]);
+      return;
+    }
+    compPriceUSD = compMarket.inputTokenPriceUSD;
+  } else {
+    // try to get TONIC price between blocks start - 1337194
```
Comment (dmelotik) [inline subgraphs/compound-forks/protocols/tectonic/src/mapping.ts:363]: should be `start-687810` right?

File: subgraphs/compound-forks/protocols/tectonic/src/mapping.ts [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+    "1.0.0",
+    "1.0.0",
+    Network.CRONOS,
+    comptroller.try_liquidationIncentiveMantissa(),
+    comptroller.try_oracle()
+  );
+  return _getOrCreateProtocol(protocolData);
+}
+
+function updateRewards(event: ethereum.Event, market: Market): void {
```
Comment (dmelotik) [inline subgraphs/compound-forks/protocols/tectonic/src/mapping.ts:269]: The naming should be cleaned up for this function

The subgraph isn't fully loaded as I am reviewing, but I am surprised there are no rewards at this point. I would do some more investigation here. Rewards and/or price might be returning 0 when not expected. 

Looking into it a little it is that the price is not being returned, when I am reviewing it should be getting the price from the market after block 1337194

**Review by dmelotik — COMMENTED** [review]
A couple small things and reward function, otherwise LGTM 🚀

**Comment by BenReilly19** [comment]
Rewards still not being queried, and logs not caught on rewards logic

File: subgraphs/compound-forks/protocols/tectonic/src/constants.ts [hunk 4]
```diff
@@ -28,7 +28,14 @@ export const TONICAddress = "0xDD73dEa10ABC2Bff99c60882EC5b2B81Bb1Dc5B2";
 
 export const tTONICAddress = "0xfe6934FDf050854749945921fAA83191Bccf20Ad";
 
-export const CRONOS_BLOCKSPERDAY = "5";
+export const RATE_IN_SECONDS = 86400;
+export const RATE_IN_SECONDS_BD = BigDecimal.fromString(
+  RATE_IN_SECONDS.toString()
+);
+
+export const CRONOS_BLOCKSPERDAY = RATE_IN_SECONDS_BD.div(
```
Comment (dmelotik) [inline subgraphs/compound-forks/protocols/tectonic/src/constants.ts:36]: to be more accurate I would make that 5.7: https://cronos.org/explorer/

File: subgraphs/compound-forks/protocols/tectonic/src/mapping.ts [hunk 5]
```diff
@@ -321,41 +330,25 @@ function updateTONICRewards(event: AccrueInterest, market: Market): void {
...
-      .div(exponentToBigDecimal(rewardDecimals));
-    log.warning("[TONIC price] Price: {}", [TonicPriceUSD.toString()]);
-  } else if (event.block.number.toI32() > 1337194) {
-    let TonicMarket = Market.load(tTONICAddress);
-    if (!TonicMarket) {
-      log.warning("[updateRewards] Market not found: {}", [tTONICAddress]);
-      return;
+  if (event.block.number.gt(BigInt.fromI32(570298))) {
+    let comptroller = Comptroller.bind(comptrollerAddr);
+    let oracle = comptroller.try_oracle();
```
[the remark this answers is not in the document]
Comment (BenReilly19) [reply subgraphs/compound-forks/protocols/tectonic/src/mapping.ts:335]: Will try to get this to work by calling lendingProtocols._priceOracle, should be possible.

File: subgraphs/compound-forks/protocols/tectonic/src/mapping.ts [hunk 6]
```diff
@@ -250,6 +250,14 @@ export function handleAccrueInterest(event: AccrueInterest): void {
 
 function getOrCreateProtocol(): LendingProtocol {
   let comptroller = Comptroller.bind(comptrollerAddr);
+  let oracle = comptroller.try_oracle();
```
[the remark this answers is not in the document]
Comment (BenReilly19) [reply subgraphs/compound-forks/protocols/tectonic/src/mapping.ts]: This was for a test, removed in newest push

File: subgraphs/compound-forks/protocols/tectonic/src/mapping.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (BenReilly19) [reply subgraphs/compound-forks/protocols/tectonic/src/mapping.ts]: It is possible, functioning subgraph has been deployed and tested.

**Comment by dmelotik** [comment]
beauty, LGTM

**Comment by this-username-is-taken** [comment]
Great work @BenReilly19! Thanks for the review @dmelotik 🙌

**Pull request merged**